src/app/features/courses/search/search.component.ts

```diff
@@ -1,20 +1,28 @@
-import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { SearchService } from '@features/services/search/search.service';
+import {ActivatedRoute} from '@angular/router';
 
 @Component({
     selector: 'app-search',
     templateUrl: './search.component.html',
     styleUrls: ['./search.component.scss']
 })
 export class SearchComponent implements OnInit {
-    @Output() searchCourses = new EventEmitter<string>();
     searchText: string;
+    searchPage: number;
 
-    constructor() { }
+    constructor(
+        private searchService: SearchService,
+        private activatedRoute: ActivatedRoute
+    ) { }
 
     ngOnInit() {
+        this.activatedRoute.queryParams.subscribe(res => {
+            this.searchPage = +res.page;
+        });
     }
 
     searchSubmit(): void {
-        this.searchCourses.emit(this.searchText);
+        this.searchService.getData(this.searchText, this.searchPage);
```

Comment:
Let's maintain search component as 'dump component', so just send all data to parent component